app/models/service_template.rb

```diff
@@ -74,6 +77,7 @@ class ServiceTemplate < ApplicationRecord
   scope :without_service_template_catalog_id,       ->         { where(:service_template_catalog_id => nil) }
   scope :with_existent_service_template_catalog_id, ->         { where.not(:service_template_catalog_id => nil) }
   scope :displayed,                                 ->         { where(:display => true) }
+  scope :public_service_templates,                  ->         { includes(:reserved_rec).select { |s| s.internal.nil? } }
```

Comment:
Should the condition be `!s.internal` to handle both nil and false values.

Reply:
This returns an array, but we want a true scope. The best I can come up is 
```
where.not(:id => ServiceTemplate.includes(:reserved_rec).joins(:reserved_rec).select { |s| s.internal == true }.collect(&:id))
```

Reply:
@bzwei 👍